File: RootTools2/src/main/java/com/rarnu/tools/neo/base/InnerFragment.java
package com.rarnu.tools.neo.base;

import android.app.Fragment;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.view.*;

public abstract class InnerFragment extends Fragment implements ViewTreeObserver.OnGlobalLayoutListener, IIntf {

    protected View innerView = null;
    protected Bundle innerBundle = null;
    public String tabTitle = null;
    public int tabIcon = -1;

    public InnerFragment() {
        super();
    }

    public InnerFragment(String tabTitle) {
        super();
        this.tabTitle = tabTitle;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (getFragmentLayoutResId() != 0) {
            innerView = inflater.inflate(getFragmentLayoutResId(), container, false);
        } else {
            innerView = getFramgmentLayoutView();
        }
        initComponents();
        initEvents();
        innerView.getViewTreeObserver().addOnGlobalLayoutListener(this);
        return innerView;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (innerBundle == null) {
            innerBundle = getArguments();
        }
        initLogic();
        if (getActivity().getActionBar() != null) {
            if (getCustomTitle() == null || getCustomTitle().equals("")) {
                if (getBarTitle() != 0) {
                    getActivity().getActionBar().setTitle(getBarTitle());
                }
            } else {
                getActivity().getActionBar().setTitle(getCustomTitle());
            }
        }
    }

    public void setNewArguments(Bundle bn) {
        innerBundle = bn;
        onGetNewArguments(bn);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        if (getActivity() == null) {
            return;
        }
        initMenu(menu);
    }



    /**
     * do NOT override this method, use @onLayoutReady instead
     */
    @Override
    public void onGlobalLayout() {
        onLayoutReady();
    }

    /**
     * override the method if you want to re-layout after system layouted
     */
    protected void onLayoutReady() {

    }

    /**
     * override the method if you do not need a layout from resource and @getFragmentLayoutResId returns 0
     */
    protected View getFramgmentLayoutView() {
        return null;
    }

    public Context getContext() {
        if (Build.VERSION.SDK_INT >= 23) {
            return super.getContext();
        } else {
            return super.getActivity();
        }
    }
}

File: RootTools2/src/main/java/com/rarnu/tools/neo/base/InnerFragment.kt
package com.rarnu.tools.neo.base

import android.app.Fragment
import android.content.Context
import android.os.Build
import android.os.Bundle
import android.view.*

/**
 * Created by rarnu on 3/23/16.
 */
abstract class InnerFragment : Fragment, ViewTreeObserver.OnGlobalLayoutListener, IIntf {

    protected var innerView: View? = null
    protected var innerBundle: Bundle? = null
    var tabTitle: String? = null
    var tabIcon = -1

    constructor() : super()
    constructor(tabTitle: String) : super() {
        this.tabTitle = tabTitle
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (getFragmentLayoutResId() != 0) {
            innerView = inflater?.inflate(getFragmentLayoutResId(), container, false)
        } else {
            innerView = getFramgmentLayoutView()
        }
        initComponents()
        initEvents()
        innerView?.viewTreeObserver?.addOnGlobalLayoutListener(this)
        return innerView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        innerBundle = innerBundle ?: arguments
        initLogic()
        if (activity.actionBar != null) {
            if (getCustomTitle() == null || getCustomTitle() == "") {
                if (getBarTitle() != 0) {
                    activity.actionBar.setTitle(getBarTitle())
                }
            } else {
                activity.actionBar.title = getCustomTitle()
            }
        }
    }

    open fun setNewArguments(bn: Bundle?) {
        innerBundle = bn
        onGetNewArguments(bn)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        if (activity == null) {
            return
        }
        initMenu(menu)
    }

    override fun getContext(): Context {
        if (Build.VERSION.SDK_INT >= 23) {
            return super.getContext()
        } else {
            return super.getActivity()
        }
    }

    /**
     * do NOT override this method, use @onLayoutReady instead
     */
    override fun onGlobalLayout() = onLayoutReady()

    /**
     * override the method if you want to re-layout after system layouted
     */
    protected open fun onLayoutReady() {
    }

    /**
     * override the method if you do not need a layout from resource and @getFragmentLayoutResId returns 0
     */
    protected open fun getFramgmentLayoutView(): View? = null
}